x-pack/plugins/alerting/server/task_runner/execution_handler.ts

```diff
@@ -369,80 +454,129 @@ export class ExecutionHandler<
-      return true;
     }
-    return false;
+
+    return executables;
   }
 
-  private isExecutableAction(action: RuleAction) {
-    return this.taskRunnerContext.actionsPlugin.isActionExecutable(action.id, action.actionTypeId, {
-      notifyUsage: true,
-    });
+  private canFetchSummarizedAlerts(action: RuleAction) {
+    const hasGetSummarizedAlerts = this.ruleType.getSummarizedAlerts !== undefined;
+
+    if (!hasGetSummarizedAlerts) {
+      this.logger.error(
+        `Skipping action "${action.id}" for rule "${this.rule.id}" because the rule type "${this.ruleType.name}" does not support alert-as-data.`
+      );
+    }
+    return hasGetSummarizedAlerts;
   }
 
-  private isExecutableActiveAlert({
-    alert,
+  private async getSummarizedAlerts({
     action,
+    ruleId,
+    spaceId,
   }: {
-    alert: Alert<AlertInstanceState, AlertInstanceContext, ActionGroupIds | RecoveryActionGroupId>;
     action: RuleAction;
+    ruleId: string;
+    spaceId: string;
   }) {
-    const alertId = alert.getId();
-    const { rule, ruleLabel, logger } = this;
-    const notifyWhen = action.frequency?.notifyWhen || rule.notifyWhen;
-
-    if (notifyWhen === 'onActionGroupChange' && !alert.scheduledActionGroupHasChanged()) {
-      if (
-        !this.skippedAlerts[alertId] ||
-        (this.skippedAlerts[alertId] &&
-          this.skippedAlerts[alertId].reason !== Reasons.ACTION_GROUP_NOT_CHANGED)
-      ) {
-        logger.debug(
-          `skipping scheduling of actions for '${alertId}' in rule ${ruleLabel}: alert is active but action group has not changed`
-        );
-      }
-      this.skippedAlerts[alertId] = { reason: Reasons.ACTION_GROUP_NOT_CHANGED };
-      return false;
+    let options;
+
+    if (isSummaryActionOnInterval(action)) {
+      const throttleMills = parseDuration(action.frequency!.throttle!);
+      const start = new Date(Date.now() - throttleMills);
+
+      options = {
+        start,
+        end: new Date(),
+        ruleId,
+        spaceId,
+      };
+    } else {
+      options = {
+        executionUuid: this.executionId,
+        ruleId,
+        spaceId,
+      };
     }
 
-    if (notifyWhen === 'onThrottleInterval') {
-      const throttled = action.frequency?.throttle
-        ? alert.isThrottled({
-            throttle: action.frequency.throttle ?? null,
-            actionHash: this.generateActionHash(action),
-          })
-        : alert.isThrottled({ throttle: rule.throttle ?? null });
+    const alerts = await this.ruleType.getSummarizedAlerts!(options);
 
-      if (throttled) {
-        if (
-          !this.skippedAlerts[alertId] ||
-          (this.skippedAlerts[alertId] && this.skippedAlerts[alertId].reason !== Reasons.THROTTLED)
-        ) {
-          logger.debug(
-            `skipping scheduling of actions for '${alertId}' in rule ${ruleLabel}: rule is throttled`
-          );
+    const total = alerts.new.count + alerts.ongoing.count + alerts.recovered.count;
+    return {
```

Comment:
nit: should we just change what's returned from `ruleType.getSummarizedAlerts` to match this instead of rewriting it?

Reply:
Changed the `getSummarizedAlerts` with https://github.com/elastic/kibana/pull/147360/commits/912217b62f34caab9f0d2004041f3d1e8baa937f, can you take a second look :)

Reply:
LGTM!